### ISSUE TITLE
Destroying a gameobject in GAMEOBJECT_POINTER_DOWN that has been set to draggable causes errors on mobile devices

### ISSUE THREAD

**Issue by sreadixl** [phaserjs/phaser#6349]
## Version

* Phaser Version: 3.55.2

* Operating system: Android 13 and iOS 15.6

* Browser: Android Chrome, iOS Safari/Chrome

## Description

If you create a GameObject, set it to be interactive and draggable (`myGo.setInteractive({draggable:true})`), set a GAMEOBJECT_POINTER_DOWN listener on it, and inside of that listener you destroy the GameObject, while your pointer is still down, errors will be thrown. This only happens on mobile devices. The stack trace is the following:
```
STACK: TypeError: Cannot read properties of undefined (reading 'target')
    at initialize.processDragMoveEvent
    at initialize.update 
    at initialize.updateInputPlugins 
    at initialize.onTouchMove 
    at HTMLCanvasElement.onTouchMove 
```

## Example Test Code

This Phaser scene will replicate the issue. This only happens on mobile devices because it affects the Touch events, on desktop the errors don't appear because they only use Mouse events.

```
export default class DragErrorExample extends Phaser.Scene {
  create() {
    this.rect = this.add.rectangle(this.scale.width / 2, this.scale.height / 2, 200, 200, 0xff0000);
    this.zone = this.add.zone(this.scale.width / 2, this.scale.height / 2, 200, 200).setInteractive({ useHandCursor: true, draggable: true });
    this.zone.on(Phaser.Input.Events.GAMEOBJECT_POINTER_DOWN, () => this.handleClick(this.zone), this);
  }

  handleClick (obj) {
    obj.destroy();
  }
}
```
## Additional Information

I tried disabling (`myGo.disableInteractive()` and removing interactive (`myGo.removeInteractive()`) from the GameObject prior to destroying it, and the error still occurs. I also tried turning off draggable on the GameObject (`this.input.setDraggable(myGo, false)`).

The error itself is happening on this line in the InputPlugin file: https://github.com/photonstorm/phaser/blob/v3.55.2/src/input/InputPlugin.js#L1268. `input` is null because the object has been destroyed. I don't know why the object is still in the drag list since it's been destroyed, and I don't quite get why it happens on mobile with touch events and not on desktop with mouse events. Might have something to do with the order events are called in.

**Comment by photonstorm** [comment]
Pretty sure this is already fixed in 3.60:

* `InputPlugin.disable` will now also reset the drag state of the Game Object as well as remove it from all of the internal temporary arrays. This fixes issues where if you disabled a Game Object for input during an input event it would still remain in the temporary internal arrays. This method now also returns the Input Plugin, to match `enable`. Fix #5828 (thank @natureofcode @thewaver)

**Comment by photonstorm** [comment]
Your test code works fine on iOS for me, so I'm going to close this one off as fixed in master. Feel free to re-open if you can prove it fails on a different mobile / browser set-up (against current master branch only, not 3.55)
